internal/client/compose.go

```diff
@@ -0,0 +1,188 @@
+// Package client - compose contains functions for the compose API
+// Copyright (C) 2020 by Red Hat, Inc.
+package client
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/osbuild/osbuild-composer/internal/weldr"
+)
+
+// PostComposeV0 sends a JSON compose string to the API
+// and returns an APIResponse
+func PostComposeV0(socket *http.Client, compose string) (*APIResponse, error) {
+	body, resp, err := PostJSON(socket, "/api/v0/compose", compose)
+	if resp != nil || err != nil {
+		return resp, err
+	}
+	return NewAPIResponse(body)
+}
+
+// NewComposeResponseV0 converts the response body to a status response
+func NewComposeResponseV0(body []byte) (*weldr.ComposeResponseV0, error) {
+	var response weldr.ComposeResponseV0
+	err := json.Unmarshal(body, &response)
+	if err != nil {
+		return nil, err
+	}
+	return &response, nil
+}
+
+// GetFinishedComposesV0 returns a list of the finished composes
+func GetFinishedComposesV0(socket *http.Client) ([]weldr.ComposeEntryV0, *APIResponse, error) {
+	body, resp, err := GetRaw(socket, "GET", "/api/v0/compose/finished")
+	if resp != nil || err != nil {
+		return []weldr.ComposeEntryV0{}, resp, err
+	}
+	var finished weldr.ComposeFinishedResponseV0
+	err = json.Unmarshal(body, &finished)
+	if err != nil {
+		return []weldr.ComposeEntryV0{}, nil, err
+	}
+	return finished.Finished, nil, nil
+}
+
+// GetFailedComposesV0 returns a list of the failed composes
+func GetFailedComposesV0(socket *http.Client) ([]weldr.ComposeEntryV0, *APIResponse, error) {
+	body, resp, err := GetRaw(socket, "GET", "/api/v0/compose/failed")
+	if resp != nil || err != nil {
+		return []weldr.ComposeEntryV0{}, resp, err
+	}
+	var failed weldr.ComposeFailedResponseV0
+	err = json.Unmarshal(body, &failed)
+	if err != nil {
+		return []weldr.ComposeEntryV0{}, nil, err
+	}
+	return failed.Failed, nil, nil
+}
+
+// GetComposeStatusV0 returns a list of composes matching the optional filter parameters
+func GetComposeStatusV0(socket *http.Client, uuids, blueprint, status, composeType string) ([]weldr.ComposeEntryV0, *APIResponse, error) {
+	// Build the query string
+	route := "/api/v0/compose/status/" + uuids
+	var filters []string
+
+	if len(blueprint) > 0 {
+		filters = append(filters, "blueprint="+blueprint)
+	}
+	if len(status) > 0 {
+		filters = append(filters, "status="+status)
+	}
+	if len(composeType) > 0 {
+		filters = append(filters, "type="+composeType)
+	}
+
+	if len(filters) > 0 {
+		route = route + "?" + strings.Join(filters, "&")
+	}
+
+	body, resp, err := GetRaw(socket, "GET", route)
+	if resp != nil || err != nil {
+		return []weldr.ComposeEntryV0{}, resp, err
+	}
+	var composes weldr.ComposeStatusResponseV0
+	err = json.Unmarshal(body, &composes)
+	if err != nil {
+		return []weldr.ComposeEntryV0{}, nil, err
+	}
+	return composes.UUIDs, nil, nil
+}
+
+// GetComposeTypesV0 returns a list of the failed composes
```

Comment:
Comment looks like a copy/paste error ?

Reply:
Looks like I pluralized the function name when I didn't mean to.

Reply:
Oh, I meant this part `list of the failed composes`.